### ISSUE TITLE
Not Getting any feedback after adding funds

### ISSUE THREAD

**Issue by omarnazih** [DarrenWestwood/greed#58]
After Adding Getting wallet info as follows
![image](https://user-images.githubusercontent.com/26510684/229506618-196a38f9-b3e0-4108-aaa7-038e12d16fae.png)

Then paying using the test/log feature
![image](https://user-images.githubusercontent.com/26510684/229506973-6a73e470-7088-4fa0-a065-93f8d144578c.png)

I Get No feedback as shown in the video
![image](https://user-images.githubusercontent.com/26510684/229507591-94fe613f-7095-4f66-b4ac-56d9b1393288.png)

**Comment by DarrenWestwood** [comment]
> Then paying using the test/log feature

Hi @omarnazih, the test bench functionality will not work after the [latest changes](https://github.com/DarrenWestwood/greed/pull/50), to change to a polling system. We are currently working on fixing the test bench functionality: https://github.com/DarrenWestwood/greed/pull/59

Unfortunately, in the meantime the only way to test the bot is using a small amount of bitcoin in a real transaction.
I will inform you in this issue, once the new testing system has been implemented.

**Comment by omarnazih** [comment]
> > Then paying using the test/log feature
> 
> Hi @omarnazih, the test bench functionality will not work after the [latest changes](https://github.com/DarrenWestwood/greed/pull/50), to change to a polling system. We are currently working on fixing the test bench functionality: #59
> 
> Unfortunately, in the meantime the only way to test the bot is using a small amount of bitcoin in a real transaction. I will inform you in this issue, once the new testing system has been implemented.

Thank you for the reply @DarrenWestwood , Do I need to change the callback url in the blockonomics platform too ?

**Comment by DarrenWestwood** [comment]
Yes, with the latest version, the Blockonomics Callback URL should be set as follows:
`https://www.blockonomics.co/api/test_callback?secret=YOUR_SECRET` (replacing your secret to match the config file as per [the updated readme](https://github.com/DarrenWestwood/greed/blob/master/README.md))

The bitcoin functionality no longer requires a public url from your bot. The new blockonomics callback url is used to compare secrets on Blockonomics and your bot.

**Comment by omarnazih** [comment]
> Yes, with the latest version, the Blockonomics Callback URL should be set as follows: `https://www.blockonomics.co/api/test_callback?secret=YOUR_SECRET` (replacing your secret to match the config file as per [the updated readme](https://github.com/DarrenWestwood/greed/blob/master/README.md))
> 
> The bitcoin functionality no longer requires a public url from your bot. The new blockonomics callback url is used to compare secrets on Blockonomics and your bot.

Great Thanks! @DarrenWestwood

**Comment by omarnazih** [comment]
@DarrenWestwood Even With real payments, Shall user get feedback or not.

**Comment by omarnazih** [comment]
@DarrenWestwood And IS there old version I Can go back too that has this feature working ?

**Comment by DarrenWestwood** [comment]
> Even With real payments, Shall user get feedback or not.

Hi @omarnazih, real payments are working correctly. The user will get notified when the payment is first received and when the payment is confirmed on the bitcoin blockchain. The user account will get credited when the payment is confirmed.

> And IS there old version I Can go back too that has this feature working ?

You can use the previous version which uses the old callback file. This version requires your bot to have a public URL as described in it's readme: https://github.com/DarrenWestwood/greed/tree/126d60d94a5cad865d0192f6e319c6a486a9fc90

The [latest development](https://github.com/DarrenWestwood/greed/pull/60) version, still in progress, is working with the Test Bench functionality. You can use this version to test the functionality, however I still recommend using the current [master](https://github.com/DarrenWestwood/greed) version and testing with small real payment when launching your production store. This development version will be released soon (less than 2 weeks), which will work with the Blockonomics test bench, and be production ready.

My apologies for any inconvenience, not having this feature working has caused

**Comment by DarrenWestwood** [comment]
The latest version has been updated to also work correctly with the Blockonomics [Test Bench](https://www.blockonomics.co/merchants#/logs). Feel free to try out the new version:
https://github.com/DarrenWestwood/greed

**Comment by omarnazih** [comment]
Thanks @DarrenWestwood I'll try right away.

**Comment by omarnazih** [comment]
Everything works Just fine Now - Small note (I had to do some adjustments to timestamp column in BtcTransaction Table) As i'm using Postgresql Database I guess you don't face this problem in sqllite.
